ocean_provider/validation/algo.py

```diff
@@ -283,68 +283,78 @@ def validate(self):
+                    not did_to_trusted_algo_dict[algorithm_did].get("containerSectionChecksum")
+            ):
+                self.error = "The trusted algorithm dict is not valid, missing " \
+                             "filesChecksum and/or containerSectionChecksum."
+
         except KeyError:
+            did_to_trusted_algo_dict = {}
             self.error = (
                 "Some algos in the publisherTrustedAlgorithms don't have a did."
             )
-            return False
 
-        for trusted_algorithm in trusted_algorithms:
-            algo = get_asset_from_metadatastore(
-                get_metadata_url(), trusted_algorithm["did"]
+        if not self.error:
+            trusted_algo_dict = did_to_trusted_algo_dict[algorithm_did]
+            algo_ddo = get_asset_from_metadatastore(
+                get_metadata_url(), trusted_algo_dict["did"]
             )
-            service = algo.get_service(ServiceTypes.METADATA)
-
-            filesChecksum = hashlib.sha256(
+            service = algo_ddo.get_service(ServiceTypes.METADATA)
+            files_checksum = hashlib.sha256(
                 (
                     service.attributes["encryptedFiles"]
                     + json.dumps(service.main["files"])
                 ).encode("utf-8")
             ).hexdigest()
 
-            if (
-                trusted_algorithm.get("filesChecksum")
-                and filesChecksum != trusted_algorithm["filesChecksum"]
-            ):
+            if files_checksum != trusted_algo_dict["filesChecksum"]:
                 self.error = (
-                    f"filesChecksum for algo with did {algo.did} does not match"
+                    f"filesChecksum for algorithm with did {algo_ddo.did} does not match"
                 )
-                return False
 
-            containerSectionChecksum = hashlib.sha256(
+            container_section_checksum = hashlib.sha256(
                 (json.dumps(service.main["algorithm"]["container"])).encode("utf-8")
             ).hexdigest()
 
-            if (
-                trusted_algorithm.get("containerSectionChecksum")
-                and containerSectionChecksum
-                != trusted_algorithm["containerSectionChecksum"]
-            ):
-                self.error = f"containerSectionChecksum for algo with did {algo.did} does not match"
-                return False
+            if container_section_checksum != trusted_algo_dict["containerSectionChecksum"]:
+                self.error = f"containerSectionChecksum for algorithm with did {algo_ddo.did} does not match"
 
-        return True
+        return bool(self.error)
+
+    def validate_algo(self):
+        """Validates algorithm details that allow the algo dict to be built."""
+        algorithm_meta = self.data.get("algorithmMeta")
+        algorithm_did = self.data.get("algorithmDid")
+        if not algorithm_did and not algorithm_meta:
```

Comment:
same here. The whole point with the validator is that these imbricated complex ifs are not necessary. Can you please return False whenever you set an error and redo this part? I'll re-review after.